Automattic/jetpack
# Plan Checks: store less info locally and simplify logic

## Proposed changes:

Until now, the `jetpack_active_plan` option included everything that was retrieved from the API. That made for a really big option, with a lot of data that we do not actually need to use in the codebase today (namely, the `available` part of the feature list).

With this PR, we remove that array from the data we store in the local option.

Of note, we were using the `available` part of the feature list in one place: starting in #19762, we introduced logic that would allow us to display an upgrade banner in the block editor, when site owners would try to add a block that was not available with their plan. The banner would tell in what plan the feature was available.
That logic relied on the `available` array from the `jetpack_active_plan` option, only to check for eligibility on WoA sites.
~Since WoA sites currently include all blocks, we do not currently need to gate access to any blocks on WoA, and thus we can remove that logic.~
Some WoA sites, namely eCommerce trial sites, do not support all blocks so we need that logic to remain. We should, however, be able to rely on the `get_minimum_plan_for_feature` method for WoA sites as well.

> [!WARNING]
> Before we merge this, we want to be absolutely sure `available` isn't used anywhere.

### Other information:


## Jetpack product discussion

See discussion here: p55Cj4-3q6-p2#comment-3642

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

> [!NOTE]
> I would recommend testing this on all 3 environments: WordPress.com Simple, WoA, and self-hosted.
> 
> Since it makes changes to data that is used across the products, I would also recommend diving through the different usages of the `Current_Plan` class in the monorepo, as well as usages of the `jetpack_active_plan` option.

* With different plans, go to Posts > Add New and attempt to insert paid blocks such as the Video block or the Calendly.
   * When your site does not support the block, a banner should be displayed.
